data/virt_autotest/alp_host_bridge_init.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+set -e -x
+#
+# PREPARATION
+# --------------------
+. /etc/os-release
+# Variables Definition
+# --------------------
+BRIDGEIF_NAME="br0"
+ACTIVE_DEVICE=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+CONNECTION_NAME=$(nmcli -f DEVICE,CONNECTION dev status| grep ${ACTIVE_DEVICE} | awk '{print $4}')
+ACTIVE_CONNECTION="Wired connection ${CONNECTION_NAME}"
+ACTIVE_UUID=$(nmcli -g UUID,DEVICE con | grep ${ACTIVE_DEVICE} | cut -d: -f1)
+IPV4_ADDRESS=$(nmcli -g IP4.ADDRESS con show "${ACTIVE_UUID}")
+IPV4_GATEWAY=$(nmcli -g IP4.GATEWAY con show "${ACTIVE_UUID}")
+IPV4_DNS_A=$(nmcli -g IP4.DNS con show "${ACTIVE_UUID}"| awk '{print $1}')
+IPV4_DNS_B=$(nmcli -g IP4.DNS con show "${ACTIVE_UUID}"| awk '{print $3}')
+
+# Function Definitions
+# --------------------
+# Add Host Bridge Network Interface via nmcli
+setup_host_bridge_network_interface() {
+
+nmcli con add type bridge \
+    con-name ${BRIDGEIF_NAME} ifname ${BRIDGEIF_NAME} \
+    autoconnect yes ipv4.method manual \
+    ipv4.address ${IPV4_ADDRESS} \
+    ipv4.gateway ${IPV4_GATEWAY} \
+    ipv4.dns ${IPV4_DNS_A} +ipv4.dns ${IPV4_DNS_B} \
+
+nmcli con add type bridge-slave \
+    con-name ${BRIDGEIF_NAME}-slave \
+    ifname ${ACTIVE_DEVICE} master ${BRIDGEIF_NAME}
+}
+
+# Enable Host Bridge Network Interface via nmcli
+enable_host_bridge_network_interface() {
+
+nmcli con up ${BRIDGEIF_NAME}
+
+# refer to bsc#1208005 for more details
+nmcli con down "${ACTIVE_CONNECTION}"
+}
```

Comment:
How about adding some  checks after bringing up the connection, eg check br0 is up with the desired ip, and ping openqa_url is successful?

Reply:
Good point! we have the same idea together! will add the basic check after bringing up the connection.

Reply:
Short update, add new function check_activate_network_interface() to check with br0 network interface.